pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -1771,6 +1778,21 @@ func (r *ReconcileClusterDeployment) deleteStaleProvisions(provs []*hivev1.Clust
 	}
 }
 
+// deleteOldFailedProvisions deletes the failed provisions which are more than 7 days old
+func (r *ReconcileClusterDeployment) deleteOldFailedProvisions(provs []*hivev1.ClusterProvision, cdLog log.FieldLogger) {
+	cdLog.Infof("Deleting failed provisions which are more than 7 days old")
```

Comment:
This should be debug level since it is run on every reconcile of an installed ClusterDeployment.